File: medical-backend/medical-system/src/main/java/com/medical/system/entity/Appointment.java
package com.medical.system.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 预约记录实体
 *
 * @author 刘柏城
 * @date 2025-11-06
 */
@Data
@TableName("appointment")
public class Appointment implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long patientId;

    private Long doctorId;

    private LocalDate appointmentDate;

    private String appointmentTime;

    private String department;

    private String symptoms;

    private Integer status;

    private String cancelReason;

    private String diagnosis;

    private String prescription;

    private BigDecimal fee;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableField(exist = false)
    private String doctorName;

    @TableField(exist = false)
    private String patientName;

    @TableField(exist = false)
    private Long patientUserId;
}

File: medical-frontend/src/api/modules/appointment.ts
import http from '@/api/http';

export interface Appointment {
    id?: number;
    patientId?: number;
    doctorId: number;
    doctorName?: string; // Transient
    patientName?: string; // Transient
    patientUserId?: number; // Transient
    department: string;
    appointmentDate: string;
    appointmentTime: string;
    symptoms: string;
    status?: number; // 0: Pending, 1: Completed, 2: Cancelled
    createTime?: string;
}

export const appointmentApi = {
    create: (data: Appointment) => http.post('/appointment', data),
    cancel: (id: number) => http.put(`/appointment/${id}/cancel`),
    complete: (id: number) => http.put(`/appointment/${id}/complete`),
    list: (params: { pageNum: number; pageSize: number; keyword?: string }) =>
        http.get<any, { records: Appointment[]; total: number }>('/appointment/list', { params })
};
